File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        if not root:
            return root
        
        ans, q = [], [root]
        while q:
            nq, level = [], []
            for node in q:
                level.append(node.val)
                if node.children:
                    nq.extend(node.children)
            if level:
                ans.append(level)
            q = nq
        return ans

File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> ans = new ArrayList<>();
        
        List<Node> q = new ArrayList<>();
        if(root != null){
            q.add(root);
        }
        
        while(q.size() > 0){
            List<Node> nq = new ArrayList<>();
            
            List<Integer> level = new ArrayList();
            for(Node node : q){
                level.add(node.val);
                
                for(Node child : node.children){
                    nq.add(child);
                }
            }
            q = nq;
            ans.add(level);
        }
        
        return ans;
    }
}